firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/cobs.txt

```diff
@@ -0,0 +1,63 @@
+Scenario: The cobs::decode method correctly decodes encoded buffers
+
+ GIVEN("The Util COBS::decode function")
```

Comment:
The tests for `COBS::decode` need to be updated so that the first THEN checks the return status code, like what you did for `COBS::encode`. In general, both functions will need to have tests triggering all possible distinct ways that error code returns can be triggered